app/src/main/java/org/mozilla/focus/shortcut/IconGenerator.kt

```diff
@@ -72,16 +72,16 @@ class IconGenerator {
             canvas.drawColor(ContextCompat.getColor(context, R.color.add_to_homescreen_icon_background))
 
             // Then draw the foreground
-            return drawCharacterOnBitmap(context, character, bitmap)
+            return drawCharacterOnBitmap(context, character, bitmap, textSize)
         }
 
-        private fun drawCharacterOnBitmap(context: Context, character: Char, bitmap: Bitmap): Bitmap {
+        private fun drawCharacterOnBitmap(context: Context, character: Char, bitmap: Bitmap, textSize: Float?): Bitmap {
             val canvas = Canvas(bitmap)
 
             val paint = Paint()
 
             val textSize = TypedValue.applyDimension(
-                    TypedValue.COMPLEX_UNIT_DIP, TEXT_SIZE_DP, context.resources.displayMetrics)
+                    TypedValue.COMPLEX_UNIT_DIP, textSize ?: TEXT_SIZE_DP, context.resources.displayMetrics)
```

Comment:
You could set the default value for the textSize in the top most function right in the parameter list:
> fun generate...(..., textSize : Float = TEXT_SIZE_DP) {

With that you can remove the `?` from all other functions so that they do not have to deal with null values.